lib/daru/index.rb

```diff
@@ -143,6 +149,11 @@ def self._load data
 
       Daru::Index.new(h[:relation_hash].keys)
     end
+
+    def factor_out
```

Comment:
I can't find any usage of this method in your code. What exactly do you plan to make it do?